rubix-spi/src/main/java/com/qubole/rubix/spi/fop/ObjectPool.java

```diff
@@ -86,8 +86,8 @@ public void registerHost(String host, int socketTimeout, int connectTimeout)
     Poolable<T> freeObject;
     if (subPool.getObjectQueue().size() == 0) {
       // increase objects and return one, it will return null if reach max size
-      int added = subPool.increaseObjects(this.config.getDelta());
-      if (added == 0) {
+      int totalObjects = subPool.increaseObjects(this.config.getDelta());
```

Comment:
so now we are not checking if totalObjects>0 and the additional connections are not created successfully?

Reply:
No, we are checking if there is atleast 1 connection which is open in the pool. `totalCount` in the ObjectPartitionPool keeps that count of total connections. As long as there is a connection active (may or may not be available right now to service request) we can expect that the request will be served